File: app/schemas/todo.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

# Todo 기본 스키마
class TodoBase(BaseModel):
    title: str
    is_completed: bool = False

# Todo 생성 스키마
class TodoCreate(TodoBase):
    pass

# Todo 업데이트 스키마
class TodoUpdate(BaseModel):
    title: Optional[str] = None
    is_completed: Optional[bool] = None

# Todo 응답 스키마
class Todo(TodoBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        orm_mode = True

File: admin-panel/types/todo.ts
// types/todo.ts
export interface Todo {
  id: number;
  title: string;
  priority: string;
  is_completed: boolean; // 'completed'를 'is_completed'로 변경
  created_at?: string;
  updated_at?: string;
  user_id?: number;
}

export interface TodoCreate {
  title: string;
  priority: string;
  is_completed?: boolean; // 'completed'를 'is_completed'로 변경
}

export interface TodoUpdate {
  title?: string;
  priority?: string;
  is_completed?: boolean; // 'completed'를 'is_completed'로 변경
}
